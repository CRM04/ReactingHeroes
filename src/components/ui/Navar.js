import React, { useState, useContext } from 'react'
import { useHistory } from "react-router-dom";
import { Transition } from "@headlessui/react";
import { NavLink } from 'react-router-dom';
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

export const Navbar = () => {
    const [isOpen, setOpen] = useState(false);
    const { user, dispatch } = useContext(AuthContext);
    const history = useHistory();

    const handleLogOut = () => {
        dispatch({
            type: types.logout
        });
        history.replace("/login");
    }

    return (
        <div className="">
            <nav className="bg-navBar shadow-2xl" >
                <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 ">
                    <div className="flex sm:justify-between lg:justify-start">
                        <div className="flex items-center">
                            <div className="">
                                <img
                                    className="h-14 w-auto"
                                    src="../assets/heroes/logo.png"
                                    alt="Workflow"
                                />
                            </div>
                            <div className="hidden md:flex justify-between">
                                <div className="ml-10 flex items-baseline space-x-4">
                                    <NavLink
                                        activeClassName="bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium"
                                        className="hover:bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium"
                                        exact
                                        to="/marvel"
                                    >
                                        Marvel
                                    </NavLink>

                                    <NavLink
                                        activeClassName="bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium"
                                        className="hover:bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium"
                                        exact
                                        to="/dc"
                                    >
                                        DC
                                    </NavLink>

                                    <NavLink
                                        activeClassName="bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium"
                                        className="hover:bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium"
                                        exact
                                        to="/search"
                                    >
                                        Search
                                    </NavLink>
                                </div>
                                <div className=" flex items-baseline space-x-4">
                                    <p className="hover:bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium">
                                        {user.name}
                                    </p>
                                    <button
                                        className="hover:bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium"
                                        onClick={handleLogOut}
                                    >
                                        Logout
                                    </button>
                                </div>
                            </div>
                        </div>
                        <div className="-mr-2 flex md:hidden">
                            <button
                                onClick={() => setOpen(!isOpen)}
                                type="button"
                                className="inline-flex items-center justify-center pt-1 pb-1 pr-2 pl-2 rounded-md text-gray-400 hover:text-white hover:bg-gray-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-800 focus:ring-white"
                                aria-controls="mobile-menu"
                                aria-expanded="false"
                            >
                                <span className="sr-only">Open main menu</span>
                                {
                                    !isOpen ? <svg
                                        className="block h-6 w-6"
                                        xmlns="http://www.w3.org/2000/svg"
                                        fill="none"
                                        viewBox="0 0 24 24"
                                        stroke="currentColor"
                                        aria-hidden="true"
                                    >
                                        <path
                                            strokeLinecap="round"
                                            strokeLinejoin="round"
                                            strokeWidth="2"
                                            d="M4 6h16M4 12h16M4 18h16"
                                        />
                                    </svg> : <svg
                                        className="hidden h-6 w-6"
                                        xmlns="http://www.w3.org/2000/svg"
                                        fill="none"
                                        viewBox="0 0 24 24"
                                        stroke="currentColor"
                                        aria-hidden="true"
                                    >
                                        <path
                                            strokeLinecap="round"
                                            strokeLinejoin="round"
                                            strokeWidth="2"
                                            d="M6 18L18 6M6 6l12 12"
                                        />
                                    </svg>
                                }


                            </button>
                        </div>
                    </div>
                </div>
                <Transition
                    show={isOpen}
                    enter="transition ease-out duration-100 transform"
                    enterFrom="opacity-0 scale-95"
                    enterTo="opacity-100 scale-100"
                    leave="transition ease-in duration-75 transform"
                    leaveFrom="opacity-100 scale-100"
                    leaveTo="opacity-0 scale-95"
                >
                    {(ref) => (
                        <div className="md:hidden" id="mobile-menu">
                            <div className="px-2 pt-2 pb-3 space-y-1 sm:px-3">
                                <NavLink
                                    activeClassName="bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium"
                                    className="hover:bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium"
                                    exact
                                    to="/marvel"
                                >
                                    Marvel
                                    </NavLink>

                                <NavLink
                                    activeClassName="bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium"
                                    className="hover:bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium"
                                    exact
                                    to="/dc"
                                >
                                    DC
                                    </NavLink>

                                <NavLink
                                    activeClassName="bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium"
                                    className="hover:bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium"
                                    exact
                                    to="/search"
                                >
                                    Search
                                    </NavLink>
                                <button
                                    className="hover:bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium"
                                    onClick={handleLogOut}
                                >
                                    Logout
                                    </button>
                                <p className="hover:bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium">
                                    {user.name}
                                </p>
                            </div>
                        </div>
                    )}

                </Transition>
            </nav>

            {/* <header className="bg-white shadow">
                <div className="max-w-7xl mx-auto py-6 px-4 sm:px-6 lg:px-8">
                    <h1 className="text-3xl font-bold text-gray-900">Dashboard</h1>
                </div>
            </header>
            <main>
                <div className="max-w-7xl mx-auto py-6 sm:px-6 lg:px-8">                    
                    <div className="px-4 py-6 sm:px-0">
                        <div className="border-4 border-dashed border-gray-200 rounded-lg h-96"></div>
                    </div>
                </div>
            </main> */}
        </div >
    )
}
