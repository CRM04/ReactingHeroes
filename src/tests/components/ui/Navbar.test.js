import { mount } from "enzyme";
import { MemoryRouter, Router } from "react-router-dom";
import { Navbar } from '../../../components/ui/Navar';
import { AuthContext } from "../../../auth/AuthContext";
import '@testing-library/jest-dom';
import { types } from "../../../types/types";

describe('<Navbar/> test suit', () => {

    const historyMock = {
        listen: jest.fn(),
        location: {},
        push: jest.fn(),
        replace: jest.fn(),
        createHref: jest.fn()
    }

    const context = {
        user: {
            logged: true,
            name: 'Cris'
        },
        dispatch: jest.fn()
    }

    const wrapper = mount(
        <AuthContext.Provider value={context}>
            <MemoryRouter>
                <Router history={historyMock}>
                    <Navbar />
                </Router>
            </MemoryRouter>
        </AuthContext.Provider>
    );

    afterEach(() => {
        jest.clearAllMocks();
    });

    test('should the component correctly', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('#userNameNav').text()).toBe('Cris');
    });


    test('should log out, clear the localstorage and go to the login', () => {
        wrapper.find('#logout').prop('onClick')();
        
        expect(context.dispatch).toHaveBeenCalledWith({
            type: types.logout
        });

        expect( historyMock.replace ).toHaveBeenCalledWith('/login');
        
    });

});