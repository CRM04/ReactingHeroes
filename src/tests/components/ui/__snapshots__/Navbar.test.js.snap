// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar/> test suit should the component correctly 1`] = `
<div
  className=""
>
  <nav
    className="bg-navBar shadow-2xl"
  >
    <div
      className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 "
    >
      <div
        className="flex sm:justify-between lg:justify-start"
      >
        <div
          className="flex items-center"
        >
          <div
            className=""
          >
            <img
              alt="Workflow"
              className="h-14 w-auto"
              src="../assets/heroes/logo.png"
            />
          </div>
          <div
            className="hidden md:flex justify-between"
          >
            <div
              className="ml-10 flex items-baseline space-x-4"
            >
              <a
                aria-current={null}
                className="hover:bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium"
                onClick={[Function]}
              >
                Marvel
              </a>
              <a
                aria-current={null}
                className="hover:bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium"
                onClick={[Function]}
              >
                DC
              </a>
              <a
                aria-current={null}
                className="hover:bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium"
                onClick={[Function]}
              >
                Search
              </a>
            </div>
            <div
              className=" flex items-baseline space-x-4"
            >
              <p
                className="hover:bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium"
                id="userNameNav"
              >
                Cris
              </p>
              <button
                className="hover:bg-gray-700 text-white px-3 py-2 rounded-md text-sm font-medium"
                id="logout"
                onClick={[Function]}
              >
                Logout
              </button>
            </div>
          </div>
        </div>
        <div
          className="-mr-2 flex md:hidden"
        >
          <button
            aria-controls="mobile-menu"
            aria-expanded="false"
            className="inline-flex items-center justify-center pt-1 pb-1 pr-2 pl-2 rounded-md text-gray-400 hover:text-white hover:bg-gray-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-offset-gray-800 focus:ring-white"
            onClick={[Function]}
            type="button"
          >
            <span
              className="sr-only"
            >
              Open main menu
            </span>
            <svg
              aria-hidden="true"
              className="block h-6 w-6"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 6h16M4 12h16M4 18h16"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
